### DISCUSSION TITLE
Multiphase simulations with grain tracker "Material has no property named: L"

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#18515] [Q&A Modules: Phase field] [answered]
Hi everyone,
I am attempting to simulate nucleation and grain growth in a dual phase material. Currently, in my input file, I have defined order parameters eta1 and eta2 for the two phases, which evolve according to the Allen-Cahn equation. To incorporate grain growth, as soon as I incorporate grain tracker, I am seeing an error "**Material has no property named: L**". I realize that mobility is required for the Allen-Cahn equation for the grains. I attempted various methods such as defining a generic material with L/ defining L as a global parameter. I also tried to define grains as gr0,gr1 etc. and using the Allen-Cahn kernel as:

```
 [./dgr0dt]
    type = TimeDerivative
    variable = gr0
    [../]

  [./ACBulk1_gr0]
    type = AllenCahn
    variable = gr0
    args = 'c'
    mob_name = L1
    f_name = F
  [../]
```

All these methods haven't worked. I'd be grateful if someone could highlight a mistake in the current approach. A pseudo-code is attached.

Warm regards,
Ritam Chatterjee

[L_error_test.txt](https://github.com/idaholab/moose/files/6925937/L_error_test.txt)

**Comment by dschwen** [comment, marked as answer]
Do you want to simulate a poly crystalline material with many grains where each grain can be one of two phases (and an arbitrary orientation)?

Check out `ACGrGrMulti` and see if that gets you anywhere. It is a multi-phase multi-grain grand potential model.

https://mooseframework.inl.gov/moose/source/kernels/ACGrGrMulti.html

@laagesen can probably elaborate further if necessary.

**Reply by ritamcj** [reply]
Thank you Daniel, this is really helpful. I will read Aagesen et al. (2018) and then try implement.

**Reply by dschwen** [reply]
By "implement" you mean write an input file? Because the model is already coded up in MOOSE as far as I can see.

**Reply by ritamcj** [reply]
Yes I will add it to my input file and make modifications in the code if necessary.

**Reply by ritamcj** [reply]
I just wanted to update that "Material has no property named: L" occurred because I had declared a velocity gradient tensor in FiniteStrainUObasedCP as 'L', which caused a mixup of variable names. Silly mistake but I'm glad I noticed that.
